### ISSUE TITLE
pxVmStackPopInstance() from VM returns nullptr (0x0)

### ISSUE THREAD

**Issue by JaXt0r** [GothicKit/phoenix-shared-interface#6]
When I register TA_MIN, it should return the following values:

```
FUNC VOID TA_Min (var c_npc self, var int start_h, var int start_m, var int stop_h, var int stop_m, VAR func state, VAR string waypoint) { };
```
I implemented the pop() logic inside PxCs.Test and get all of the data except the instance which will always be 0x0.

![image](https://user-images.githubusercontent.com/120568393/231801578-e610ee95-caf9-4bc7-a724-7ae144849f42.png)

(Code: https://github.com/GothicKit/phoenix-csharp-interface/blob/main/PxCs.Tests/PxVmTest.cs#L74)

As the other data is poped() right, I assume the order is correct?

```c#
        public static void TA_MIN(IntPtr vmPtr)
        {
            var waypoint = PxVm.pxVmStackPopString(vmPtr); // OCR_HUT_33 --> correct
            var action = PxVm.pxVmStackPopInt(vmPtr); // 5903 --> correct
            var stop_m = PxVm.pxVmStackPopInt(vmPtr);
            var stop_h = PxVm.pxVmStackPopInt(vmPtr);
            var start_m = PxVm.pxVmStackPopInt(vmPtr);  // 30 --> correct
            var start_h = PxVm.pxVmStackPopInt(vmPtr);  // 22 --> correct
            var npc = PxVm.pxVmStackPopInstance(vmPtr); // --> error. Shouldn't be 0

            Assert.True(npc != IntPtr.Zero, "Npc is IntPtr.Zero.");
        }
```

What could drive this error?

**Comment by lmichaelis** [comment]
Can you provide me with a stack trace of the VM? I just added an API on main for printing it: `pxVmPrintStackTrace`? Please print it before popping anything off the stack.

**Comment by JaXt0r** [comment]
I found the issue. I forgot to call ```pxVmSetGlobalSelf()``` before the actual VM call.

Nevertheless the Stack printing feature will not work as it prints to std::cerr which won't be shared with the managed C# side. I think we would need to return a heap allocated string. But question is if it's worth the effort, as the system was right:
npcInstance == 0x0 if self() wasn't set. ;-)

**Comment by lmichaelis** [comment]
That issue was fixed upstream in [d6566d5](https://github.com/GothicKit/phoenix/commit/d6566d5db3fb58f4ce5d83e4dda2dd7925c84dc0).
